app/views/play/level/tome/Spell.coffee

```diff
@@ -81,6 +81,9 @@ module.exports = class Spell
           }
         }
       """
+    if @language is 'cpp' and not @languages[@language]
+      # TODO: do some transforms here to handle most common sample code differences between JavaScript and C++
+      @languages.cpp = @languages.java
```

Comment:
What is this line doing?

Reply:
It's a placeholder for any further automatic transforms we want to do. But yeah, I guess this should be coming from `javascript`.